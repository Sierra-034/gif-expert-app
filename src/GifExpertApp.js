import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";

const GifExpertApp = () => {

    const [ categories, setCategories ] = useState(['One Punch', 'Samurai X', 'Dragon ball']);

    // const handleAdd = () => {
    //     setCategories( prevState => [ ...prevState, 'Scooby Doo' ] );
    // };

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory />
            <hr/>
            {/* <button onClick={ handleAdd }>Agregar</button> */}
            <ol>
                { categories.map(category => {
                    return <li key={ category }>{ category }</li>
                }) }
            </ol>
        </>
    );
};

export default GifExpertApp;