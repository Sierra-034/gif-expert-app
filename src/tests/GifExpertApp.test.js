import { shallow } from 'enzyme';
import GifExpertApp from "../GifExpertApp";

describe('Pruebas sobre <GifExpertApp />', () => {
    test('Coincide con el snapshot', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorÃ­as', () => {
        const categories = ['halo', 'veritasium'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});
