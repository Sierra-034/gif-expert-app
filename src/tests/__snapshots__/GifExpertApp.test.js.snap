// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifExpertApp /> Coincide con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas sobre <GifExpertApp /> Debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="halo"
      key="halo"
    />
    <GifGrid
      category="veritasium"
      key="veritasium"
    />
  </ol>
</Fragment>
`;
